recipes/openmm/meta.yaml

```diff
@@ -0,0 +1,97 @@
+{% set name = "openmm" %}
+{% set version = "7.4.0" %}
+{% set build = 0 %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/openmm/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: cb7ddd923d6f402cf2723b3c14509915abad328505e8ce401ac3131706dabeff
+
+build:
+  number: {{ build }}
+  skip: True  # [py2k or win]
+  missing_dso_whitelist:
+  - "*/libcuda.*"  # [linux]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}  # [linux]
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - cmake
+    - jom  # [win]
+
+  host:
+    - python
+    - fftw
+    - swig
+    - pthread-stubs
+    - lxml
+    - numpy
+    - cython
+    # needed for Python wrappers
+    - doxygen 1.8.14
```

Comment:
Why is this version needed?

Reply:
@peastman: Do you recall which Doxygen versions work correctly with the OpenMM Swig system?

Reply:
I don't know if this was in response to a particular problem.  In general, every time we change versions of Doxygen or SWIG there's a good chance something will break.  So when possible it's best to pin the build tools to particular versions we know work.  We also should be pinning SWIG, since we know there's a bug in recent versions that breaks the OpenMM documentation (https://github.com/sphinx-doc/sphinx/issues/6565).

Reply:
Latest `doxygen` is `v1.8.16`, which breaks the build. `1.8.14` works correctly, that's why it's pinned. Maybe we can relax the pinning a bit (`<=`?). Regarding SWIG, we are not building the docs in this recipe, so we are not observing that bug, I'd say.

Reply:
It would be better to be as permissive as we can, @peastman, since we could cause package conflicts if we pin to a single version.

Reply:
For runtime dependencies we should be permissive.  Build time dependencies don't cause problems for users, and we know from past experience that changes to the build tools have a very high probability of breaking things.